java-frontend/src/test/java/org/sonar/java/SonarComponentsTest.java

```diff
@@ -728,52 +729,44 @@ void fileCanBeSkipped_returns_false_when_the_file_is_a_generated_file() throws A
         checkFactory
       )
     );
+    SensorContext contextMock = mock(SensorContext.class);
+    sonarComponents.setSensorContext(contextMock);
+
     InputFile inputFile = new GeneratedFile(Path.of("non-existing-generated-file.java"));
 
     assertThat(sonarComponents.fileCanBeSkipped(inputFile)).isFalse();
   }
 
   @Test
-  void fileCanBeSkipped_always_returns_false_when_skipUnchangedFiles_is_false() throws ApiMismatchException {
+  void fileCanBeSkipped_always_returns_false_when_skipUnchangedFiles_is_false() throws ApiMismatchException, IOException {
 
     SonarComponents sonarComponents = mock(SonarComponents.class, CALLS_REAL_METHODS);
+    SensorContext contextMock = mock(SensorContext.class);
+    sonarComponents.setSensorContext(contextMock);
+
     when(sonarComponents.canSkipUnchangedFiles()).thenReturn(false);
     InputFile inputFile = mock(InputFile.class);
 
     assertThat(sonarComponents.fileCanBeSkipped(inputFile)).isFalse();
-    verify(inputFile, times(0)).status();
   }
 
   @Test
-  void fileCanBeSkipped_always_returns_true_when_skipUnchangedFiles_is_true_and_file_status_is_same() throws ApiMismatchException {
+  void fileCanBeSkipped_returns_false_when_inputFileStatusIsDifferentFromSame() throws ApiMismatchException {
```

Comment:
Let's see if we need to adapt the logic of this test by first disabling the cache for `fileCanBeSkipped`to fallback on the input file status